Edits cannot be saved in the EU Conversions page

**Describe the bug**
After editing the information for a row in the EU Conversions page, the following error is produced when attempting to save the edits.

![Image](https://github.com/user-attachments/assets/78e8f86c-a066-457f-a86e-dfe37e899c07)

**To Reproduce**
Steps to reproduce the behavior:
1. Go to the EU Conversions page of the OpenDCS Web Client.
2. Select a row to work with.
3. Edit the information for the selected row.
4. Click the Save button.
5. Observe that the error will be produced.

**Expected behavior**
After editing a row in the EU conversions page, the edits should be saved. If there is incorrect information with the edited information, the user should be informed so that corrections can be made.

In retesting this issue, saving edits to a unit conversion still could not be saved. 

Status 400 is produced.

The following stack trace was collected:
```
10-Apr-2025 16:54:16.712 WARNING [https-jsse-nio-8443-exec-9] org.opendcs.odcsapi.res.AppExceptionMapper.handle There was an error.  Please contact your sys admin.
	org.opendcs.odcsapi.dao.DbException: Unable to store Unit Converter list
		at org.opendcs.odcsapi.res.DatatypeUnitResources.postEUConv(DatatypeUnitResources.java:421)
		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
		at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:146)
		at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:189)
		at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
		at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
		at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
		at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
		at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
		at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:256)
		at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
		at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
		at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
		at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
		at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
		at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
		at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:235)
		at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684)
		at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:394)
		at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:346)
		at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:358)
		at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:311)
		at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:205)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:660)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687)
		at org.apache.catalina.authenticator.SingleSignOn.invoke(SingleSignOn.java:312)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:360)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:893)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1789)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Thread.java:750)
	Caused by: decodes.db.DatabaseException: Cannot set ID on UnitConverterDb to 3194433126: ID on this object already set to 3194432126
		at decodes.db.IdDatabaseObject.setId(IdDatabaseObject.java:87)
		at decodes.sql.UnitConverterIO.addNew(UnitConverterIO.java:245)
		at decodes.sql.SqlDatabaseIO.insertUnitConverter(SqlDatabaseIO.java:1016)
		at org.opendcs.odcsapi.res.DatatypeUnitResources.postEUConv(DatatypeUnitResources.java:416)
		... 49 more
```

--- a/opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/DatatypeUnitResources.java
+++ b/opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/DatatypeUnitResources.java
@@ -412,6 +412,7 @@ public final class DatatypeUnitResources extends OpenDcsResource
 			if(ucId != null)
 			{
 				dbIo.deleteUnitConverter(ucId);
+				unitConverterDb.forceSetId(DbKey.NullKey);
 			}
 			dbIo.insertUnitConverter(unitConverterDb);
 			return Response.status(HttpServletResponse.SC_CREATED).entity(map(unitConverterDb)).build();

